google-apis-core/lib/google/apis/core/api_command.rb

```diff
@@ -161,6 +162,7 @@ def set_api_client_header
           xgac = "gl-ruby/#{RUBY_VERSION} gdcl/#{munged_client_version}"
           xgac = old_xgac.empty? ? xgac : "#{old_xgac} #{xgac}"
           header.delete_if { |k, v| k.downcase == 'x-goog-api-client' }
+          xgac += invocation_id_header if options.add_invocation_id_header
```

Comment:
We probably need a space to separate the invocation ID clause from the rest of the header.

Reply:
+1, there needs to be a space here.